testonly/integration/mapenv.go

```diff
@@ -64,6 +65,10 @@ func NewMapEnvFromConn(addr string) (*MapEnv, error) {
 
 // NewMapEnv creates a fresh DB, map server, and client.
 func NewMapEnv(ctx context.Context) (*MapEnv, error) {
+	if provider := testdb.Default(); !provider.IsMySQL() {
```

Comment:
Do you need to repeat the !IsMySQL check here, when you've done it in map_test.go?
There, you skip the test, and would not go on to execute this.
Or is mapenv.go used in other tests too?

Reply:
This check makes sense here because it verifies the preconditions necessary to use MapEnv. 
Individual tests may also choose to `t.Skip()`, but in terms of functional correctness, a check here seems sensible.   It also provides a nice error when the preconditions for `MapEnv` are not meet rather than spewing out db locked garbage.

`NewMapEnv` is used by KeyTransparency.

Reply:
OK, it's fine.